### ISSUE TITLE
Bug: a GraalVM environment initialized with SDKMAN overrides `node` binary

### ISSUE THREAD

**Issue by pshirshov** [sdkman/sdkman-cli#804]
This is a common problem with all the environment management tools.

Let's assume I have a local GraalVM environment initialized with SDKMAN:

```
work/tut/sdkman-test via ☕ v11.0.8 on ☁️  eu-west-1
❯ cat .sdkmanrc
# Enable auto-env through the sdkman_auto_env config
# Add key=value pairs of SDKs to use below
java=20.2.0.r11-grl
```

This would hopelessly override my `node` binary:

```
work/tut/test via ☕ v11.0.8 on ☁️  eu-west-1
❯ which node
/Users/pavel/.sdkman/candidates/java/current/bin/node
```

This is very inconvenient because I have a heterogeneous project which requires both `node` and `java` but GraalVM's `node` is like 2 times slower than original one.

I believe there must be a way to disable overrides for some binaries. E.g. `.sdkmanrc` may contain something like:

```
java=20.2.0.r11-grl
ignore=node
```

**Comment by helpermethod** [comment]
Hi @pshirshov!

This can easily be fixed by an `alias`

    alias node=/path/to/your/node/binary

**Comment by pshirshov** [comment]
Nope, subprocesses wouldn't honor `alias`es. Moreover, my primary shell is `zsh` while my automation scripts are mostly in bash and scala.

**Comment by pshirshov** [comment]
For those who may experience the same problem: I made an alternative dependency manager: https://github.com/7mind/binlink

**Comment by marc0der** [comment]
This problem could be easily fixed with a post-install hook in our [sdkman-hooks](https://github.com/sdkman/sdkman-hooks) service, that is exactly what the hooks are for. For instance, we could simply take away executable rights on node (and any other) binary. We could even remove those files all together.

@pshirshov just a quick observation about the above comment. You are completely entitled to building your own tool to rival the likes of SDKMAN, jenv or others, after all this is a free and open world. However, please don't use our issues or public forums as a means of self-promotion for your own tool in the future. It's simply not cool.

We would like to see SDKMAN become better over time, so we're always open to ideas and inputs and we welcome pull requests to make this tool better for everyone to use.

**Comment by goeh** [comment]
It took me a while to figure out why my Vue.js application suddenly took 20 minutes to start instead of 20 seconds! Realized after a lot of frustration that I used ~/.sdkman/candidates/java/current/bin/node that came with the GraalVM installation I did last week.

**Comment by viniciusccarvalho** [comment]
Same here, this is kinda annoying and I wish they just removed this from sdkman. I have to remember to switch aways from graalvm everytime I need to use anything with node. 
SDKMAN is awesome, but this is kinds diminishing its value :(

**Comment by ravi9856** [comment]
I have used NVM, and put that after sdkman in .zshrc. It resolved the issue.

export SDKMAN_DIR="/Users/ravindra.kumar/.sdkman"
[[ -s "/Users/ravindra.kumar/.sdkman/bin/sdkman-init.sh" ]] && source "/Users/ravindra.kumar/.sdkman/bin/sdkman-init.sh"
export PATH="/usr/local/opt/mysql@5.7/bin:$PATH"

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

**Comment by helpermethod** [comment]
GraalVM 21.1 and future versions do not package a node binary anymore. So installing the latest version of GraalVM will solve that problem.

**Comment by deepankarmalhan** [comment]
The comment above mine worked!
